frontend/src/container/PublicProfile.jsx

```diff
@@ -62,46 +81,64 @@ export class PublicProfile extends React.Component {
+    props.dispatch(newFlashMessage(error.message))
+    props.history.push(PAGE.HOME)
+    return true
+  }
+
   getUser = async () => {
     const { props } = this
     const userId = props.match.params.userid
 
     const fetchGetUser = await props.dispatch(getAboutUser(userId))
-
-    switch (fetchGetUser.status) {
-      case 200:
-        this.setState({
-          displayedUser: {
-            ...serialize(fetchGetUser.json, serializeUserProps),
-            userId: userId
-          },
-          coverImageUrl: 'default'
-        })
-        this.buildBreadcrumbs()
-        break
-      case 400:
-        switch (fetchGetUser.json.code) {
-          case 1001:
-            props.dispatch(newFlashMessage(props.t('Unknown user')))
-            props.history.push(PAGE.HOME)
-            break
-          default:
-            props.dispatch(newFlashMessage(props.t('Error while loading user')))
-            props.history.push(PAGE.HOME)
-        }
-        break
-      default:
-        props.dispatch(newFlashMessage(props.t('Error while loading user')))
-        props.history.push(PAGE.HOME)
-    }
+    if (this.handleFetchErrors(
+      fetchGetUser,
+      [{ status: 400, code: 1001, message: props.t('Unknown user') }],
+      props.t('Error while loading user')
+    )) return
+
+    const apiUser = { ...serialize(fetchGetUser.json, serializeUserProps) }
+    this.setState({
+      displayedUser: apiUser,
+      coverImageUrl: 'default'
+    })
+    this.buildBreadcrumbs()
   }
 
+  onChangeAvatarClick = () => this.setState({ displayUploadPopup: POPUP_DISPLAY_STATE.AVATAR })
+
+  onChangeAvatarSuccess = () => this.props.dispatch(updateUserProfileAvatarName(`avatar-${Date.now()}`))
+
+  onCloseUploadPopup = () => this.setState({ displayUploadPopup: undefined })
+
   render () {
     const { props, state } = this
-
+    const changeImageEnabled = (state.displayedUser.userId === props.user.userId) || props.user.profile === PROFILE.ADMINISTRATOR
+    const avatarBaseUrl = getAvatarBaseUrl(FETCH_CONFIG.apiUrl, state.displayedUser.userId)
     return (
       <div className='tracim__content fullWidthFullHeight'>
         <div className='tracim__content-scrollview'>
+          {state.displayUploadPopup === POPUP_DISPLAY_STATE.AVATAR && (
+            <PopupUploadFile
+              label={props.t('Upload an image')}
+              uploadUrl={`${avatarBaseUrl}/avatar/raw`}
+              httpMethod='PUT'
+              color={GLOBAL_primaryColor}
+              handleClose={this.onCloseUploadPopup}
+              handleSuccess={this.onChangeImageSuccess}
+              allowedMimeTypes={ALLOWED_IMAGE_MIMETYPES}
+              maximumFileSize={MAXIMUM_AVATAR_SIZE}
+            >
+              <i className='fa fa-fw fa-arrows-alt' /> {props.t('Recommended dimensions:')} 100x100px<br />
+              <i className='fa fa-fw fa-image' /> {props.t('Maximum size: {{size}} MB', { size: MAXIMUM_AVATAR_SIZE / (1024 * 1024) })}
```

Comment:
The keys are case-sensitive, if you use `MB` here, don't forget to change at the translation files.